napari/_qt/qt_main_window.py

```diff
@@ -870,6 +870,10 @@ def _add_viewer_dock_widget(
             menu.addAction(action)
         # self.window_menu.addAction(action)
 
+        # see #3663, to fix #3624 more generally
```

Comment:
Ah ok you moved code here, that's probably fine - I'm just tracking

Reply:
Yup exactly, this change basically makes sure that that fix works for all dock widgets, not just plugin dock widgets. Still ugly, but it works.